File: content/problems/0012-integer-to-roman/solutions/greedy-symbol-table/Solution.java
class Solution {
    public String intToRoman(int num) {
        int[] values = {1000,900,500,400,100,90,50,40,10,9,5,4,1}; String[] symbols = {"M","CM","D","CD","C","XC","L","XL","X","IX","V","IV","I"};
        StringBuilder answer = new StringBuilder();
        for (int index = 0; index < values.length; index++) while (num >= values[index]) { num -= values[index]; answer.append(symbols[index]); }
        return answer.toString();
    }
}

File: content/problems/0012-integer-to-roman/solutions/greedy-symbol-table/Solution.kt
class Solution {
    fun intToRoman(input: Int): String {
        val values = intArrayOf(1000,900,500,400,100,90,50,40,10,9,5,4,1); val symbols = arrayOf("M","CM","D","CD","C","XC","L","XL","X","IX","V","IV","I")
        var num = input; val answer = StringBuilder()
        for (index in values.indices) while (num >= values[index]) { num -= values[index]; answer.append(symbols[index]) }
        return answer.toString()
    }
}
